cloud/webfaction/webfaction_facts.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python
+#
+# Gather facts from Webfaction using Ansible and the Webfaction API
+#
+# ------------------------------------------
+#
+# (c) Pascal Bach 2015, with contributions gratefully acknowledged from:
+#     * Quentin Stafford-Fraser
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: webfaction_facts
+short_description: Gather facts from a Webfaction machine
+description:
+    - Gather facts about applications, databases, websites and domains form a Webfaction machine
+author: Pascal Bach (@bachp)
+version_added: "2.0"
+notes:
+    - "You can run playbooks that use this on a local machine, or on a Webfaction host, or elsewhere, since the scripts use the remote webfaction API - the location is not important.
+    - See `the webfaction API <http://docs.webfaction.com/xmlrpc-api/>`_ for more info.
+
+options:
+
+    login_name:
+        description:
+            - The webfaction account to use
+        required: true
+
+    login_password:
+        description:
+            - The webfaction password to use
+        required: true
+
+    machine:
+        description:
+            - The machine name to use (optional for accounts with only one machine)
+        required: false
```

Comment:
add a default if `required: false` e.g. `default: null` or `default: none` in this case